### ISSUE TITLE
Rancher 2.2: Login fails for users with escaped comma in CN when using Active Directory LDAP authentication

### ISSUE THREAD

**Issue by FFock** [rancher/rancher#22192]
**Bug Report**
Exception in log when trying to login (some users may login, but Admin user cannot):

**Steps to reproduce**
Use an active directory that includes an escaped "\," (comma) in the CN.  

**Result:**
The following log output is produced (personal data replaced in quoted string)
2019/08/15 11:46:12 [ERROR] Error refreshing token principals, skipping: LDAP Re                                                                                                    sult Code 34 "Invalid DN Syntax": 0000208F: NameErr: DSID-03100225, problem 2006                                                                                                     (BAD_NAME), data 8350, best match of:
        ', Name (999999),OU=99,OU=Tool-Users,DC=domain-index,DC=com'

**Other details that may be helpful:**
CN in LDAP (ActiveDirectory) looks like "CN=LastName\, Forname (999999)"

**Environment information**
- Rancher 2.2 latest stable.
- Single install (on RHEL 7.7 with SE Linux)

Related issue: https://github.com/rancher/rancher/issues/9827

**Comment by mrajashree** [comment]
@FFock yes the dn in that request is not being escaped causing that error. But your admin should still be able to login. If admin cannot login, can you share the exact error in logs when admin tries to login?

**Comment by jiaqiluo** [comment]
I cannot reproduce the bug in v2.3.1,

Steps:
- run Rancher:v2.3.1 single install
- enable the authentication with `Active Directory`
- connect to the Active Directory management server via Microsoft Remote Desktop client
- added new users with the following variances:
	-- the first name contains `\,`
	-- the last name contains `\,`
	-- the display name contains `\,`
	-- the user login name contains `\,`

**Comment by FFock** [comment]
As menitioned in my initial report, the problem does not directly occur after initial setup and it occurs only when the comma is escaped in the CN. The problem occurs several hours after adding a user to the allowed users. Then some cache-update fails in the background due to the LDAP CN field parsing error and then no further login is possible with that user. 
In our LDAP server all user accounts have an properly escaped comma in their CN but not in any other field. The problem occurs after several hours to a few days after a user was authorized to login in Rancher. When the problem shows up, no login (even not for the admin) is possible anymore.
We are currently blocked to use Rancher 2.x. Rancher 1.6.x is working with the same LDAP server though.

**Comment by mrajashree** [comment]
@jiaqiluo Can you try this:
1. After setting up AD, add any of the test users containing comma to allowed users list
2. Login as the user with comma in dn.
3. Log back in as the admin and check that a "User" account is created on the Global->Users page for this user with comma in dn.
4. While on the Global->Users page click on the "Refresh Group Memberships" button
Then you should see error in the logs:
```
2019/11/04 19:12:30 [ERROR] Error refreshing token principals, skipping: LDAP Result Code 34 "Invalid DN Syntax": 0000208F: NameErr: DSID-03100225, problem 2006 (BAD_NAME), data 8350, best match of:
	', first,CN=xxxxx'
```

@FFock yes a call to get all user's current groups fails in the background due to incorrectly escaped dn. However the logins are still working. Do the users see any error in the UI or in the logs when the logins fail for them?

**Comment by jiaqiluo** [comment]
The bug fix for the escaped comma is validated in v2.3-head 22483d475

Results:

- the error is not showing in the log
- and the group is refreshed successfully

**Comment by FFock** [comment]
Tested with Rancher 2.3.2 again and the bug is still there:

2019/11/25 14:05:58 [INFO] Creating user for principal activedirectory_user://CN=Name\, Forename (110099),OU=06,OU=Users,DC=domain,DC=com
2019/11/25 14:05:58 [INFO] Creating globalRoleBindings for u-ux3ix4axu5
2019/11/25 14:05:58 [INFO] Creating new GlobalRoleBinding for GlobalRoleBinding grb-v729l
2019/11/25 14:05:58 [INFO] [mgmt-auth-grb-controller] Creating clusterRoleBinding for globalRoleBinding grb-v729l for user u-ux3ix4axu5 with role cattle-globalrole-user
2019/11/25 14:05:58 [ERROR] Error refreshing token principals, skipping: LDAP Result Code 34 "Invalid DN Syntax": 0000208F: NameErr: DSID-0310022E, problem 2006 (BAD_NAME), data 8350, best match of:
        ', Forename (110099),OU=06,OU=Users,DC=domain,DC=com'

**Comment by mrajashree** [comment]
@FFock the fix is not in 2.3.2, it's in 2.3.3, can you please test with 2.3.3?

**Comment by deniseschannon** [comment]
Verified fix by QA as of v2.3.3.
